src/components/LineChart/LineChart-container.tsx

```diff
@@ -26,20 +27,19 @@ const LineCharts = (props: LineChartProps): React.ReactElement => {
   ]);
   const [data, setData] = useState([10, 20, 30, 40]);
 
-
-  const filterChart = (): void => {
-    const labels: string[] = [];
-    const elements: number[] = [];
-
-    transactionsSummed.forEach((item: TransactionSummed): void => {
-      elements.push(item.totalValue);
-      labels.push(convertTimeStampToHours(item.interval));
-    });
-    setLabels(labels);
-    setData(elements);
-  };
-
   useEffect((): void => {
+    const filterChart = (): void => {
+      const labels: string[] = [];
+      const elements: number[] = [];
+
+      transactionsSummed.forEach((item: TransactionSummed): void => {
+        elements.push(item.totalValue);
+        labels.push(convertTimeStampToHours(item.interval));
+      });
+      setLabels(labels);
+      setData(elements);
+    };
```

Comment:
czy ta cała funkcja nie powinna być poza useEffectem, a w useEffect tylko się wywoływać?

Reply:
no wlasnie to jest prawidłową implementacja useEffect i nie wszędzie da sie to teraz przerobić dlatego musiałem dac disable rule